session/tlv.go

```diff
@@ -152,3 +154,186 @@ func DeserializeSession(r io.Reader) (*Session, error) {
 
 	return session, nil
 }
+
+// encodeMacaroonRecipe serializes a macaroon recipe so that it can be stored
+// as a tlv value. The serialization will follow the following pattern:
```

Comment:
If we want to make this encoding _more_ flexible. We can us a _nested_ TLV instance here. Sounds scary, but it just means adding a length prefix before the inner TLV blob, and doing TLV as normal within that.

Reply:
ok nice! i decided to go all out on the nested-tlv thing after seeing this comment. Let me know if it is too much now 😅